toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -779,12 +783,12 @@ func validateOutput(output imagecustomizerapi.Output, outputImageFile string) er
 func customizeImageHelper(buildDir string, baseConfigPath string, config *imagecustomizerapi.Config,
 	rawImageFile string, rpmsSources []string, useBaseImageRpmRepos bool, partitionsCustomized bool,
 	imageUuidStr string,
-) (map[string]diskutils.FstabEntry, error) {
+) (map[string]diskutils.FstabEntry, string, error) {
 	logger.Log.Debugf("Customizing OS")
 
-	imageConnection, partUuidToFstabEntry, err := connectToExistingImage(rawImageFile, buildDir, "imageroot", true)
+	imageConnection, partUuidToFstabEntry, osRelease, err := connectToExistingImage(rawImageFile, buildDir, "imageroot", true)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 	defer imageConnection.Close()
```

Comment:
`imageconnection.Chroot().RootDir()` has the entire disk mounted. So, it would be easy to read the `/etc/os-release` file here. And doing so would avoid adding complexity to `connectToExistingImage`.